js/Ksay.js

```
@@ -1,283 +1,322 @@
+'use strict';
+
 function Ksay() {
     var activeTab = null;
     var tabs = {};
     var blocks = {};
     var objects = {};
     var objectTypes = {};
     var elementId = 1;
-    //Get unique id
+
+    // Get an unique id
     this.generateNewID = function () {
         return 'KsayID' + elementId++;
-    }
-    //Return true if tab exists
+    };
+
+    // Return true if the tab exists
     this.tabExists = function (tab) {
-        return tabs[tab] != undefined && tabs[tab] != null;
-    }
-    //Return true if block exists
+        return !!tabs[tab];
+    };
+
+    // Return true if the block exists
     this.blockExists = function (block) {
-        return blocks[block] != undefined && blocks[block] != null;
-    }
-    //Set page title
+        return !!blocks[block];
+    };
+
+    // Set the page title
     this.setTitle = function (title) {
         document.getElementById('mainTitle').innerText = title;
-    }
-    //Add tab. Error if tab already exists;
+    };
+
+    // Add a tab. Prints an error message if the tab already exists.
     this.addTab = function (name, title) {
-        //If tab does not exist yet
         if (this.tabExists(name)) {
             console.log('Ksay: tab "' + name + '" already exists');
```

Please try throwing an exception instead of printing the message.